hack/add_queue_extensions.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+RTPLUGS_PKG="" 
+RTPLUGS_NAMES=""
+
+# RTPLUGS_PKG should include a comma seperated list of go packages
+# Each package should implement a plug of github.com/IBM/go-security-plugs/rtplugs
+# All packages will be staticaly loaded to queue proxry 
+#
+# Package names that will be defined in the podSpec runtimePlugins and those 
+# defined in the config-defaults configmap runtime-plugins parameter will be activated
+
+
+# ----------- Do not edit above this line ------------------
+
+# Comment out packages that you do not wish to load
+
+# Add the testing workload security guard plug - harmless, yet better not use in production 
+RTPLUGS_PKG="github.com/IBM/go-security-plugs/plugs/testgate"
+RTPLUGS_NAMES="testgate"
+
+# Add the sample workload security guard plug - do not use in production 
+#RTPLUGS_PKG="${RTPLUGS_PKG},github.com/IBM/go-security-plugs/plugs/rtgate"
+
+# Add workload security guard plug (under development) - use at your own risk
+# RTPLUGS_PKG="${RTPLUGS_PKG},github.com/IBM/workload-security-guard/wsgate"
+
+# Add here additinal packages that you wish to load in this template    
+# RTPLUGS_PKG="${RTPLUGS_PKG}github.com/HomerSimpson/BestSecurityEver/shutdown"
+
+
+# ----------- Do not edit below this line ------------------
+
+cd cmd/queue
+echo "------------------------"
+echo "Generating queue.go"
+mv -n queue.go queue.go.orig
+
+cat <<EOT > queue.go
+// Code generated by $0. DO NOT EDIT.
+
+package main
+
+import "os"
+
+import "github.com/IBM/go-security-plugs/qpsecurity"
+
+EOT
+
+IFS="," read -r -a PKG_ARRAY <<< ${RTPLUGS_PKG}
+for p in ${PKG_ARRAY}
+do
+  # process
+  echo import _ \"$p\"
+  echo "import _ \"$p\"" >> queue.go
+  go get $p
+done
+
+cat <<EOT >> queue.go
```

Comment:
Let's break this down to a couple of items:

1. `queue.mainWithPlugs(qpExtension)` -  moving `main.go` to a separate pkg such as "knative.dev/serving/pkg/cmd/queue" (or a pkg under it) seems like a good option 
(if everyone is on board with this move).

2. Having`qpExtension := qpsecurity.ConfigFromAnnotationFile("/podinfo/annotations", "rtplugs")`:
    `qpsecurity` is a QP specific glue pakage, so we need not have this call, it can read from "/podinfo/annotations" (if that is an agreed QP core path for the annotations) during e.Init(). It can reside in the sandbox, moving forward. It can be renamed qpextender.

3.  `defer qpExtension.Shutdown()` - as indicated other comment - is a less attractive option due to flush().